great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -360,30 +373,25 @@ def execution_engine_type(self) -> Type[ExecutionEngine]:
 
         return SqlAlchemyExecutionEngine
 
-    class Config:
-        arbitrary_types_allowed = True
-
-    def __init__(self, **kwargs) -> None:
-        super().__init__(**kwargs)
-        # validate that SQL Alchemy was successfully imported and attempt to create an engine
-        if SQLALCHEMY_IMPORTED:
-            if "connection_string" in kwargs and kwargs["connection_string"]:
+    def get_engine(self) -> sqlalchemy.engine.Engine:
+        if self.connection_string != self._cached_connection_string or not self._engine:
+            # validate that SQL Alchemy was successfully imported and attempt to create an engine
+            if SQLALCHEMY_IMPORTED:
                 try:
-                    self._engine = sqlalchemy.create_engine(kwargs["connection_string"])
+                    self._engine = sqlalchemy.create_engine(self.connection_string)
                 except Exception as e:
-                    # connection_string has passed pydantic validation,
-                    # but still fails to create a sqlalchemy engine
+                    # connection_string has passed pydantic validation, but still fails to create a sqlalchemy engine
+                    # one possible case is a missing plugin (e.g. psycopg2)
```

Comment:
Could you add `from e` to the `raise SQLDatasourceError` line?

Reply:
Also, do you want to set `self._cached_connection_string` to `None` in the except block?

Reply:
Added `from e` - thanks for that.

I think we'd rather keep the successfully cached `self._cached_connection_string` (and `self._engine`) intact in the event of a `create_engine` failure.